IONFileViewerLib/IONFLVWPreviewer.swift

```diff
@@ -0,0 +1,31 @@
+import Foundation
+import QuickLook
+import AVKit
+
+class IONFLVWPreviewer: QLPreviewControllerDataSource {
+    
+    lazy var previewItem = NSURL()
+    weak var viewController: UIViewController?
+    
+    init(viewController: UIViewController) {
+        self.viewController = viewController
+    }
+    
+    func previewMediaContent(url: URL) throws {
+        let player = AVPlayer(url: url.standardized)
+        let playerViewController = AVPlayerViewController()
+        playerViewController.player = player
+        viewController?.present(playerViewController, animated: true) {
```

Comment:
We could use guard let on `viewController` to return early or throw in case it is `null` before creating instances of `AVPlayer` and `AVPlayerController`, since on that case they wouldn't be used.

Reply:
Right, that makes sense.

My point was more about returning early and avoiding creating unnecessary instances.